www_src/pages/section-2/section-2.jsx

```diff
@@ -62,16 +66,54 @@ var Slot = React.createClass({
 });
 
 var Grid = React.createClass({
-  zoom: function (amount) {
+  zoomToPage: function (event) {
+    var midpointX = Math.floor(this.tilesPerRow / 2);
+    var midpointY = Math.floor(this.tilesPerCol / 2);
+
+    var cameraX = 0;
+    var cameraY = 0;
+
+    var isEvenWidth = this.tilesPerRow % 2 === 0;
+    var isEvenHeight = this.tilesPerCol % 2 === 0;
+
+    if (event.x > midpointX) {
+      cameraX = (event.x - midpointX) * -1 * this.slotWidth;
+
+      if (isEvenWidth) {
+        cameraX += (this.slotWidth / 2);
+      }
+    } else if (event.x <= midpointX) {
+      cameraX = (midpointX - event.x) * this.slotWidth;
+
+      if (isEvenWidth) {
+        cameraX -= (this.slotWidth / 2);
+      }
+    }
+
+    if (event.y > midpointY) {
+      cameraY = (event.y - midpointY) * -1 * this.slotHeight;
+
+      if (isEvenHeight) {
+        cameraY += (this.slotHeight / 2);
+      }
+    } else if (event.y <= midpointY) {
+      cameraY = (midpointY - event.y) * this.slotHeight;
+
+      if (isEvenHeight) {
+        cameraY -= (this.slotHeight / 2);
+      }
+    }
+
     this.setState({
-      zoom: amount
+      zoom: this.state.containerWidth / this.slotWidth,
+      cameraX: cameraX,
+      cameraY: cameraY
     });
   },
   getInitialState: function () {
     var layout = generateGrid(3, 3);
 
     return {
-      zoom: this.props.initialZoom,
```

Comment:
you're still referencing `this.state.zoom` in places, so either those need fixing, or `zoom` needs to stay in the initial state